core/src/test/java/org/infinispan/api/mvcc/repeatable_read/CacheAPIOptimisticTest.java

```diff
@@ -10,4 +10,12 @@
    protected IsolationLevel getIsolationLevel() {
       return IsolationLevel.REPEATABLE_READ;
    }
+
+   @Override
+   public void testRetainAllMethodOfEntryCollection() {
+      //pruivo.note:
+      //write-skew is not stored in ImmortalCacheEntry
+      //should we add equals() to MetadataImmortalCacheEntry and re-implement the test using it?
+      //TBH, it doesn't make much sense to expose our internal cache entries...
```

Comment:
that isn't that simple. it is invoking the ```equals()``` of our internal entries. An ```ImmortalEntry``` is not equals to a ```MetadataImmortalEntry``` or a mortal/transient entry.  I think all of ```entrySet``` code needs to be fixed to check for the ```key``` and ```value``` and not use the ```equals()```.
anyway, it isn't the scope of this PR.